godot-macros/src/util/kv_parser.rs

```diff
@@ -156,36 +155,64 @@ impl KvParser {
     }
 
     /// Handles an optional key that can occur with arbitrary tokens as the value.
-    pub fn handle_expr(&mut self, key: &str) -> ParseResult<Option<TokenStream>> {
+    ///
+    /// Returns both the key (with the correct span pointing to the attribute) and the value.    
+    /// [KvParser.span](field@KvParser::span) always points to the top of derive macro (`#[derive(GodotClass)]`).
+    pub fn handle_expr_with_key(&mut self, key: &str) -> ParseResult<Option<(Ident, TokenStream)>> {
         match self.map.remove_entry(&ident(key)) {
             None => Ok(None),
             // The `key` that was removed from the map has the correct span.
             Some((key, value)) => match value {
                 None => bail!(key, "expected `{key}` to be followed by `= expression`"),
-                Some(value) => Ok(Some(value.expr()?)),
+                Some(value) => Ok(Some((key, value.expr()?))),
             },
         }
     }
 
-    pub fn handle_usize(&mut self, key: &str) -> ParseResult<Option<usize>> {
-        let Some(expr) = self.handle_expr(key)? else {
+    /// Shortcut for [KvParser::handle_expr_with_key] which returns only the value.
+    pub fn handle_expr(&mut self, key: &str) -> ParseResult<Option<TokenStream>> {
+        match self.handle_expr_with_key(key)? {
+            Some((_key, value)) => Ok(Some(value)),
+            None => Ok(None),
+        }
+    }
+
+    pub fn handle_literal(
+        &mut self,
+        key: &str,
+        expected_type: &str,
+    ) -> ParseResult<Option<Literal>> {
+        let Some((key, expr)) = self.handle_expr_with_key(key)? else {
             return Ok(None);
         };
 
         let mut tokens = expr.into_iter();
         let Some(TokenTree::Literal(lit)) = tokens.next() else {
             return bail!(
                 key,
-                "missing value for '{key}' (must be unsigned integer literal)"
+                "missing value for '{key}' (must be {expected_type} literal)"
             );
         };
 
         if let Some(surplus) = tokens.next() {
             return bail!(
                 key,
-                "value for '{key}' must be unsigned integer literal; found extra {surplus:?}"
+                "value for '{key}' must be {expected_type} literal; found extra {surplus:?}"
             );
         }
+        Ok(Some(lit))
+    }
+
+    /// Handles a string literal (`att = "str"`).
+    pub fn handle_string(&mut self, key: &str) -> ParseResult<Option<String>> {
+        self.handle_literal(key, "String")
+            .map(|possible_literal| possible_literal.map(|lit| lit.to_string()))
+    }
```

Comment:
We didn't add this because string properties can generally be expressions.

It might be OK to support only literals in the first spec, but maybe add a comment like

> If we support expressions in the future (e.g. names of `const`s), this is no longer needed.